### PR TITLE
Refactor logging

### LINKED ISSUE #1720: Enhance and add proper logging
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
With the new infra creation added, add proper logging for better understanding.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:

This PR adds

- Logs wherever necessary
- Proper naming for PowerVS and VPC
- Consistent error format starting with `failed to`
- Add appropriate log levels, debug set at level 3
- Consistent naming of infra resources

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1720 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Refactor logging
```

### CODE REVIEW

**Review by Karthik-K-N — COMMENTED** [review]
This is one of the much needed PR, Thanks for submitting this.

/lgtm

**Review by Prajyot-Parab — APPROVED** [review]

**Comment by Amulyam24** [comment]
@Karthik-K-N @Prajyot-Parab , what are your thoughts on changing the log level to 3 by default?  
By default if user doesn't set `v` flag, we will be missing important log messages with the current changes right?

**Comment by Karthik-K-N** [comment]
> @Karthik-K-N @Prajyot-Parab , what are your thoughts on changing the log level to 3 by default? By default if user doesn't set `v` flag, we will be missing important log messages with the current changes right?

If we go as per capi standards the logs are not for users its for sre/developers https://cluster-api.sigs.k8s.io/developer/logging#logging.
So we can keep info/generic logs at level 1, debug information at 3 and more detailed info around 5 v levels.

So we can update the doc in developer section of the book about setting v level to higher one if the user is interested in debugging things.

Meanwhile we should spend some time in adding more events and condition with user perspective in mind.

**Comment by Amulyam24** [comment]
> > @Karthik-K-N @Prajyot-Parab , what are your thoughts on changing the log level to 3 by default? By default if user doesn't set `v` flag, we will be missing important log messages with the current changes right?
> 
> If we go as per capi standards the logs are not for users its for sre/developers https://cluster-api.sigs.k8s.io/developer/logging#logging. So we can keep info/generic logs at level 1, debug information at 3 and more detailed info around 5 v levels.
> 
> So we can update the doc in developer section of the book about setting v level to higher one if the user is interested in debugging things.
> 
> Meanwhile we should spend some time in adding more events and condition with user perspective in mind.

Thank you for the input @Karthik-K-N, will refactor the logs accordingly.

**Comment by dharaneeshvrd** [comment]
@Amulyam24 errors appended during reconcileDelete are not printed in controller log, there is no way to know it as well, please see whether we can add a log message [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/5707edab77fcec70024cc8bebe1523a2b0e59091/controllers/ibmpowervscluster_controller.go#L319)

**Comment by mkumatag** [comment]
@Amulyam24 can you please resolve the conflicts and update the PR?

**Comment by Amulyam24** [comment]
> @Amulyam24 can you please resolve the conflicts and update the PR?

PR is ready for review, PTAL @mkumatag @Karthik-K-N

File: cloud/scope/powervs_machine.go [hunk 1]
```diff
@@ -397,30 +397,28 @@ func (m *PowerVSMachineScope) Name() string {
...
 
 	cosClient, err := m.createCOSClient()
 	if err != nil {
-		m.Error(err, "failed to create cosClient")
-		return "", fmt.Errorf("failed to create cosClient %w", err)
+		return "", fmt.Errorf("failed to create COS client %w", err)
 	}
 	key := m.bootstrapDataKey()
-	m.Info("bootstrap data key", "key", key)
+	m.V(3).Info("Bootstrap data key details", "key", key)
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_machine.go:408]: Its just a name, so may be
```suggestion
	m.V(3).Info("Bootstrap data key name", "key", key)
```

File: cloud/scope/powervs_cluster.go [hunk 2]
```diff
@@ -976,12 +977,12 @@ func (s *PowerVSClusterScope) ReconcileVPC() (bool, error) {
 	// TODO(karthik-k-n): create a generic cluster scope/service and implement common vpc logics, which can be consumed by both vpc and powervs
 
 	// create VPC
-	s.Info("Creating a VPC")
+	s.V(3).Info("Creating a VPC")
 	vpcDetails, err := s.createVPC()
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_cluster.go:981]:
```suggestion
	vpcID, err := s.createVPC()
```

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**